### DISCUSSION TITLE
JIT Compiler ignores tailwind.config from imported npm module

### DISCUSSION THREAD

**Discussion by gplusdotgr** [tailwindlabs/tailwindcss#9146] [Help]
We developed a 'shared' npm module that will be imported in Vue.js. 

Both the 'host' applications and the 'shared' (guest) npm module use tailwind, but they have slightly different configurations and are using different classes. 

As the host application bundles and compiles tailwind (JIT), it does not take into account the classes and config used in the shared dependency.

Is there a way to make tailwind aware of that, or should we just pre-compile the CSS of the shared dependency individually? 

```
 ─ frontend-app
 ─ tailwind.config.js
 ── node_modules
 ─── @shared-components
 ──── widget
 ──── tailwind.config.js
 ```

```
import Widget from '@shared-components/widget'
```

**Comment by wongjn** [comment]
You could add the shared component to your project `tailwind.config.js`, with a preset:

```js
const sharedPreset = require('@shared-components/tailwind.preset');

module.exports = {
  content: [
    'node_modules/@shared-components/widget/*.vue',
    // Your project content globs.
  ],
  presets: [sharedPreset],
  // …
};
```

Otherwise if there would be conflicts, then I might go for precompiling your shared component CSS.

Edit: use presets as per [comment](https://github.com/tailwindlabs/tailwindcss/discussions/9146#discussioncomment-3440553).

**Comment by adamwathan** [comment]
I would read about [presets](https://tailwindcss.com/docs/presets), might be what you need for pulling in the shared config. Then just make sure your `content` configuration is scanning the templates in your shared library.

**Reply by gplusdotgr** [reply]
@adamwathan thanks Adam! about that last part, `Then just make sure your content configuration is scanning the templates in your shared library.`
does this mean adding something like `node_modules/@shared-components/*.vue` to `content` ?  - cause I tried this and it did not work (is it supposed to?)

**Reply by wongjn** [reply]
Something like `node_modules/@shared-components/**/*.vue` should work – `/*` is only in the intermediate directory so `/**/*.vue` searches in sub directories too, so this would include your `widget` folder.

**Reply by gplusdotgr** [reply]
that indeed worked, and extending the config wasnt even necessary ...
